SeQuant/core/context.cpp

```diff
@@ -31,15 +31,17 @@ set_scoped_default_context(const Context& ctx) {
   return detail::set_scoped_implicit_context(ctx);
 }
 
-Context::Context(Vacuum vac, IndexSpaceMetric m, BraKetSymmetry bks,
+Context::Context(Vacuum vac, IndexSpaceRegistry isr, IndexSpaceMetric m, BraKetSymmetry bks,
                  SPBasis spb, std::size_t fdio)
     : vacuum_(vac),
+      idx_space_reg_(std::make_shared<IndexSpaceRegistry>(isr)),
       metric_(m),
       braket_symmetry_(bks),
       spbasis_(spb),
       first_dummy_index_ordinal_(fdio) {}
 
 Vacuum Context::vacuum() const { return vacuum_; }
+std::shared_ptr<IndexSpaceRegistry> Context::index_space_registry() const {return idx_space_reg_;};
```

Comment:
@cmasteran1 you must install `pre-commit` git hooks and use it throughout. Autoformatting is the only sane way ...